### PR TITLE
fix(jsonrpc): replace RADType::Unknown with RADType::HttpGet for data requests older than the WIP0019 activation epoch

### PR DESCRIPTION
**Pull request by drcpu-github**
Currently, fetching a data request completed before activation of WIP0019 results in that data request being displayed with an `Unknown` RADRetrieve kind due to the changes made to retrieval types. Obviously, this is not correct, these data requests were correct at the time and them being shown as invalid could result in incorrect on-chain analysis and can also be considered a problem for tools suchs as the explorer or external wallets. 

Before this patch, an RPC call to below data requests results in the RADRetrieve kind to be shown as `Unknown`.
`./witnet node transaction a065dbeb6d8efa142555e82dab51fe0e5c31a985e43ea55949a01f261a143fcc`
`{"jsonrpc":"2.0","result":{"blockHash":"392b6dbda4e0ac7cb73a17c40e4a01ad30e96e1302d370dcc653c037b701aabf","confirmed":true,"transaction":{"DataRequest":{"body":{"dr_output":{"collateral":22000000000,"commit_and_reveal_fee":100,"data_request":{"aggregate":{"filters":[],"reducer":3},"retrieve":[{"kind":"Unknown","script":[...],"url":"https://www.bitstamp.net/api/ticker/"},{"kind":"Unknown","script":[...],"url":"https://api.coindesk.com/v1/bpi/currentprice.json"}],"tally":{"filters":[],"reducer":3},"time_lock":1604853236},"min_consensus_percentage":51,"witness_reward":100,"witnesses":100},"inputs":[...],"outputs":[...]},"signatures":[...]}}},"id":"1"}`

After this patch, the RADRetrieve kind is correctly shown as `HTTP-GET`.
`./witnet node transaction a065dbeb6d8efa142555e82dab51fe0e5c31a985e43ea55949a01f261a143fcc`
`{"jsonrpc":"2.0","result":{"blockHash":"392b6dbda4e0ac7cb73a17c40e4a01ad30e96e1302d370dcc653c037b701aabf","confirmed":true,"transaction":{"DataRequest":{"body":{"dr_output":{"collateral":22000000000,"commit_and_reveal_fee":100,"data_request":{"aggregate":{"filters":[],"reducer":3},"retrieve":[{"kind":"HTTP-GET","script":[...],"url":"https://www.bitstamp.net/api/ticker/"},{"kind":"HTTP-GET","script":[...],"url":"https://api.coindesk.com/v1/bpi/currentprice.json"}],"tally":{"filters":[],"reducer":3},"time_lock":1604853236},"min_consensus_percentage":51,"witness_reward":100,"witnesses":100},"inputs":[...],"outputs":[...]},"signatures":[...]}}},"id":"1"}`

Obviously, the best way to solve this would be to patch the actual database. However, this seems rather invasive and I believe this patch which essentially acts as a simple translation layer is the next best thing.

Let me know what you think!

### CODE REVIEW

**Review by tmpolaczyk — COMMENTED** [review]
Makes sense, and it works.

Note that this does not fix other JSON-RPC methods such as `getBlock` or `dataRequestReport`, which could also return a `RADType::Unknown`, but we can open an issue to fix them later.

File: node/src/actors/json_rpc/json_rpc_methods.rs [hunk 1]
```diff
@@ -806,8 +823,37 @@ pub async fn get_transaction(hash: Result<(Hash,), jsonrpc_core::Error>) -> Json
...
+            let new_transaction = match transaction {
+                Transaction::DataRequest(mut dr_txn) => {
+                    // Create Active WIPs
+                    let active_wips = ActiveWips {
+                        active_wips: active_upgrades.unwrap(),
+                        block_epoch,
+                    };
+                    if !active_wips.wip0019() {
+                        // Replace the Unknown RADType with RADType::HttpGet for all epochs before the activation of WIP0019
+                        dr_txn.body.dr_output.data_request.retrieve = dr_txn
```
[the remark this answers is not in the document]
Comment (drcpu-github) [reply node/src/actors/json_rpc/json_rpc_methods.rs:836]: That's a nice simplification. I'm not that fluent in Rust. 😉

File: node/src/actors/json_rpc/json_rpc_methods.rs [hunk 2]
```diff
@@ -806,8 +823,37 @@ pub async fn get_transaction(hash: Result<(Hash,), jsonrpc_core::Error>) -> Json
                     return Err(internal_error(e));
                 }
             };
+
+            let new_transaction = match transaction {
+                Transaction::DataRequest(mut dr_txn) => {
+                    // Create Active WIPs
```
[the remark this answers is not in the document]
Comment (drcpu-github) [reply node/src/actors/json_rpc/json_rpc_methods.rs:829]: Yeah, I consider doing that, but I liked the async sends to be bundled. In any case, I don't have a strong preference for either, so I just applied it.

**Comment by drcpu-github** [comment]
> Makes sense, and it works.
> 
> Note that this does not fix other JSON-RPC methods such as `getBlock` or `dataRequestReport`, which could also return a `RADType::Unknown`, but we can open an issue to fix them later.

I actually checked `dataRequestReport` before and it does not contain any source information, so it does not suffer from that issue. I forgot about `getBlock`. I'll merge that in this patch too. Seems silly to create a new patch and commit for the same issue.

**Review by tmpolaczyk — APPROVED** [review]
I see, the `dataRequestReport` CLI command actually calls `getTransaction`, so that's also fixed.

In `getBlock`, it's very important that the `RADType` is changed after the `txns_hashes` field is calculated, because otherwise the calculated transactions hashes will be wrong. So you can add a short comment about that, rebase with master, and merge.

**Comment by drcpu-github** [comment]
Rebased and merged commits. I left the `getBlock` and `getTransaction` as separate commits for clarity. I don't have write access so someone else will have to merge.

**Comment by tmpolaczyk** [comment]
Merged, thank you!

**Pull request merged**